slang.h

```diff
@@ -517,6 +517,7 @@ extern "C"
         SLANG_EXECUTABLE,           ///< Executable (for hosting CPU/OS)
         SLANG_SHARED_LIBRARY,       ///< A shared library/Dll (for hosting CPU/OS)
         SLANG_HOST_CALLABLE,        ///< A CPU target that makes the compiled code available to be run immediately
+        SLANG_TARGET_COUNT_OF,
```

Comment:
Hmm... I'm still not on board for the `COUNT_OF` naming (I'd prefer `SLANG_TARGET_COUNT`). There's the other issue of whether these are generally safe to expose in public API - what would a client use these for such that they'd maintain source/binary compatibility when building or running against new versions of Slang?